### PR TITLE
Refactor carousel scroll logic

### PR DESCRIPTION
**Pull request by nobuti**
This PR implements #7328.

We are fixing the binding, and the logic for checking the scroll and shadows has been refactored. Also, it includes some styles fixes for Safari, and some optimizations for shadows in order to show the buttons completely.

Please @xavijam take a look and let me now if shadows are ok now.

### CODE REVIEW

File: app/assets/stylesheets/editor-3/editor.css.scss [hunk 1]
```diff
@@ -43,6 +44,7 @@
 }
 .CDB-Dashboard-canvas {
   @include block-shadow();
+  flex: 1;
```
Comment (xavijam) [inline app/assets/stylesheets/editor-3/editor.css.scss]: Check mixins please :)

File: app/assets/stylesheets/editor-3/menu.css.scss [hunk 2]
```diff
@@ -9,6 +9,7 @@
   box-sizing: border-box;
   padding: $baseSize * 2 0;
   background: $cBlue;
+  flex-shrink: 0;
```
Comment (xavijam) [inline app/assets/stylesheets/editor-3/menu.css.scss]: Same for this one :)

File: app/assets/stylesheets/editor-3/editor.css.scss (hunk 1 above)
Comment (piensaenpixel) [inline app/assets/stylesheets/editor-3/editor.css.scss]: One thing, we only are using CDB for CartoAssets, i think in this case we should change to Editor-Dashboard-...
@xavijam ?

Comment (xavijam) [inline app/assets/stylesheets/editor-3/editor.css.scss]: Yep, it would be worth it, could we change it @nobuti please?

Comment (nobuti) [inline app/assets/stylesheets/editor-3/editor.css.scss]: @xavijam @piensaenpixel that class name is coming from deep-insight. There we use `a tope` the  CDB prefix for classes, not sure about the implications.

Comment (xavijam) [inline app/assets/stylesheets/editor-3/editor.css.scss]: We can't touch it then. 💃

**Comment by xavijam** [comment]
Tested in staging, :+1:. Looks good man!

**Comment by xavijam** [comment]
@piensaenpixel, would you like to add something about styles :)?

**Comment by piensaenpixel** [comment]
na it's ok for me

**Pull request merged**